File: web/di_core/domain/Response/Authentication/SessionInfo.ts
/*
 * @author: tvc12 - Thien Vi
 * @created: 5/30/21, 11:06 PM
 */

export class SessionInfo {
  key!: string;
  value!: string;
  domain!: string;
  timeoutInMS!: number;
  maxAge!: number;
  path = '/';
}

File: server/src/main/scala/co/datainsider/caas/user_profile/domain/user/SessionInfo.scala
package co.datainsider.caas.user_profile.domain.user

@SerialVersionUID(1L)
case class SessionInfo(
    key: String,
    value: String,
    domain: String,
    maxAgeInMs: Long,
    createdAt: Option[Long],
    path: String = "/"
) {
  val maxAge = maxAgeInMs / 1000L

  val expiredAt = createdAt.getOrElse(System.currentTimeMillis()) + maxAgeInMs
}

/**
  * @author anhlt
  */
object SessionInfo {
  val ATTR_USER = "user"
  val ATTR_ORGANIZATION_ID = "organization_id"
  val ATTR_ORGANIZATION = "organization"

}
